designs/kwok-provider.md

```diff
@@ -0,0 +1,51 @@
+# Karpenter KwoK Cloud Provider - v0.34+
+
+The kubernetes-sigs/karpenter repo uses a simulated kubernetes environment to test code changes. Users must use a cloud provider of their choice to test their changes, which can be costly and creates an unreasonable barrier to entry for contributors. Related issue: https://github.com/kubernetes-sigs/karpenter/issues/895.
+
+Creating a Cloud Provider neutral implementation of Karpenter within Karpenter-core requires creating a set of CP neutral examples of instance types and pricing that captures common tenets across CPs. 
+
+This is meant to be a living document to in-depth go through the decisions on how kubernetes-sigs/karpenter models its fake instance types. 
+
+## Proposal
+
+### Instance Types
+The full set of instance types for the KwoK Cloud Provider should be at least the cartesian product of the following dimensions, generating 60 instance types across 4 zones, totaling `12 * 3 * 2 = 64` in-memory Cloud Provider instance types.
+
+```md
```

Comment:
I recommend making fake Linux nodes and fake Windows nodes, which means that `kubernetes.io/os` would be relevant too.

We can support fake GLADOS nodes too if we want, and then verify that neither Windows nor Linux Pods get scheduled there.

Reply:
Yeah currently it'll support linux and windows, so we should be good here. It would be pretty easy to add in new OS keys in the future too!